src/main/java/java/lang/StringBuilder.java

```diff
@@ -611,14 +611,22 @@ public int lastIndexOf(String str, int fromIndex) {
     }
 
     /**
-     * @diffblue.noSupport
+     * @diffblue.fullSupport
+     * @diffblue.untested
      */
     @Override
     public StringBuilder reverse() {
         // super.reverse();
         // return this;
-        CProver.notModelled();
-        return CProver.nondetWithNullForNotModelled();
+        int size = this.length();
+        if (size < 2)
+          return this;
+        String tmp = this.toString();
+        CProverString.delete(this, 0, size);
+        for (int i=size-1; i>=0; --i) {
+            this.append(CProverString.charAt(tmp, i));
```

Comment:
:bulb: this will create a lot of intermediary strings which could be avoided using one nondetString and then assume on each character, but otherwise it looks fine.